SNAP-1110: set up as XCFramework

igor-snapyr opened the pull request:
Seems to be impossible to configure xcodeproject to immediately build xcframework, so I made a script buildxcframework.sh that combines frameworks to xcframework

paulwsmith commented on buildxcframework.sh:
```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+echo "Creating directory 'builtframework' where XCFramework will be stored"
+mkdir builtframework
+mkdir builtframework/ .archives
```
Extraneous line to remove?

igor-snapyr replied:
Yep, removed it, tested, was unnecessary

paulwsmith reviewed (commented):
I get the following when I run this, and don't end up with an archive. Do we need to make updates to the project settings to address this? 

I get similar errors for macOS, Apple TV Simulator and Apple tvOS.

(using an Intel Mac, macOS 12.2.1, Xcode 13.3)


```
Archiving for macOS
2022-03-16 14:05:03.644 xcodebuild[23826:3296145] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionSentinelHostApplications for extension Xcode.DebuggerFoundation.AppExtensionHosts.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
2022-03-16 14:05:03.644 xcodebuild[23826:3296145] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionPointIdentifierToBundleIdentifier for extension Xcode.DebuggerFoundation.AppExtensionToBundleIdentifierMap.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild archive -scheme Snapyr -archivePath builtframework/.archives/Snapyr-macosx.xcarchive -sdk macosx SKIP_INSTALL=NO

User defaults from command line:
    IDEArchivePathOverride = /Users/paul/dev/snapyr/snapyr-ios-sdk/builtframework/.archives/Snapyr-macosx.xcarchive
    IDEPackageSupportUseBuiltinSCM = YES

Build settings from command line:
    SDKROOT = macosx12.3
    SKIP_INSTALL = NO

--- xcodebuild: WARNING: Using the first of multiple matching destinations:
{ platform:macOS, arch:x86_64h, id:4B21D23B-217A-5E9E-A372-A28F19462989, error:My Mac doesn’t support any of Snapyr.framework’s architectures. You can set Snapyr.framework’s Architectures build setting to Standard Architectures to support My Mac. }
{ platform:iOS, id:dvtdevice-DVTiPhonePlaceholder-iphoneos:placeholder, name:Any iOS Device }
{ platform:iOS Simulator, id:dvtdevice-DVTiOSDeviceSimulatorPlaceholder-iphonesimulator:placeholder, name:Any iOS Simulator Device }
{ platform:iOS Simulator, id:249C255A-CEC1-4B20-A8B4-0BDC3557E662, OS:15.4, name:iPad (9th generation) }
{ platform:iOS Simulator, id:3A29153D-2383-4B6B-8D5F-945104845391, OS:15.4, name:iPad Air (4th generation) }
{ platform:iOS Simulator, id:F25D6F6A-A802-47AF-BC47-3B9A4A992A9A, OS:15.4, name:iPad Air (5th generation) }
{ platform:iOS Simulator, id:A5F85EFF-D1BC-455C-94E2-969539B64F29, OS:15.4, name:iPad Pro (9.7-inch) }
{ platform:iOS Simulator, id:4775EE4C-AB9F-4EBC-9D5B-86830BBD5E47, OS:15.4, name:iPad Pro (11-inch) (3rd generation) }
{ platform:iOS Simulator, id:E9C646F0-F77B-4127-9DC7-374F6DB8A408, OS:15.4, name:iPad Pro (12.9-inch) (5th generation) }
{ platform:iOS Simulator, id:048A7719-A095-4455-92C8-8217C85C7910, OS:15.4, name:iPad mini (6th generation) }
{ platform:iOS Simulator, id:DB25D478-AA55-4043-8882-CEE001863C4F, OS:15.4, name:iPhone 8 }
{ platform:iOS Simulator, id:ACF7A64C-34F0-4BDB-8F16-1AC53D0EEE15, OS:15.4, name:iPhone 8 Plus }
{ platform:iOS Simulator, id:59338CA4-BA97-4ADD-92B9-240C894EE728, OS:15.4, name:iPhone 11 }
{ platform:iOS Simulator, id:07AD5812-F8EA-4F26-8A6D-13721DA5EEF1, OS:15.4, name:iPhone 11 Pro }
{ platform:iOS Simulator, id:60B9454D-8577-4E20-9883-211EA1575366, OS:15.4, name:iPhone 11 Pro Max }
{ platform:iOS Simulator, id:648E12CB-C546-4052-A0A9-276C8CC4D55C, OS:15.4, name:iPhone 12 }
{ platform:iOS Simulator, id:A6D274DF-BB1C-421A-AC84-DEF507ACE31B, OS:15.4, name:iPhone 12 Pro }
{ platform:iOS Simulator, id:FF03D3E6-2C2C-4012-9082-44510CA6582F, OS:15.4, name:iPhone 12 Pro Max }
{ platform:iOS Simulator, id:48100AEC-3DCC-43B1-86DD-D05AC6866CC4, OS:15.4, name:iPhone 12 mini }
{ platform:iOS Simulator, id:214F07C6-FA68-45EF-9E5D-BD6D049D8A84, OS:15.4, name:iPhone 13 }
{ platform:iOS Simulator, id:6DA03A1A-6CD3-44EB-8666-87288A39083B, OS:15.4, name:iPhone 13 Pro }
{ platform:iOS Simulator, id:991C85DE-3E3F-4552-B11D-B30238379EF8, OS:15.4, name:iPhone 13 Pro Max }
{ platform:iOS Simulator, id:31AE86AC-E459-4E97-8241-47167FFED0D3, OS:15.4, name:iPhone 13 mini }
{ platform:iOS Simulator, id:BEFF3FA3-66E0-44A7-9F5E-1939BE858735, OS:15.4, name:iPhone SE (2nd generation) }
{ platform:iOS Simulator, id:65F31D8E-847E-42C6-8338-476A81571B64, OS:15.4, name:iPhone SE (3rd generation) }
{ platform:iOS Simulator, id:1065ED0B-E48A-462C-8E9E-3E08E896F798, OS:15.4, name:iPod touch (7th generation) }
xcodebuild: error: Failed to build project Snapyr with scheme Snapyr.
	Reason: The run destination My Mac is not valid for Archiving the scheme 'Snapyr'.
	Recovery suggestion: My Mac doesn’t support any of Snapyr.framework’s architectures. You can set Snapyr.framework’s Architectures build setting to Standard Architectures to support My Mac.
```

paulwsmith reviewed (commented):
A couple other notes:

1. Until now, we've run `make archive` to build + zip the framework (see https://github.com/snapyrautomation/snapyr-ios-sdk/blob/main/Makefile). The resulting .zip is what we publish to our Github releases page. Let's update the makefile to reference this new script? 
2. Do we need to make any updates to our Cocoapods configuration for this to keep working correctly through Cocoapods?

igor-snapyr commented:
```
I get the following when I run this, and don't end up with an archive. Do we need to make updates to the project settings to address this?

I get similar errors for macOS, Apple TV Simulator and Apple tvOS.

(using an Intel Mac, macOS 12.2.1, Xcode 13.3)
```
Yes, the error is related to the project settings, SUPPORTED_PLATFORMS in project settings only contained iphoneos, fixed.

igor-snapyr commented:
```
A couple other notes:

1. Until now, we've run make archive to build + zip the framework (see https://github.com/snapyrautomation/snapyr-ios-sdk/blob/main/Makefile). The resulting .zip is what we publish to our Github releases page. Let's update the makefile to reference this new script?
2. Do we need to make any updates to our Cocoapods configuration for this to keep working correctly through Cocoapods?
```
1. Updated the makefile, and added zip to script
2. I think no, there's no option like in project settings, only deployment targets

paulwsmith commented:
lgtm 👍
